js/main.js

```diff
@@ -16,40 +16,65 @@ $journalEntry.addEventListener('submit', createEntry);
 function createEntry(event) {
   event.preventDefault();
   var entryObj = {};
-  var title = $journalEntry.elements.title.value;
-  var photoUrl = $journalEntry.elements.photoUrl.value;
-  var notes = $journalEntry.elements.notes.value;
-  entryObj.title = title;
-  entryObj.photoUrl = photoUrl;
-  entryObj.notes = notes;
-  entryObj.entryId = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(entryObj);
-  $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $entryList.prepend(renderEntry(entryObj));
-  $journalEntry.reset();
+  var title = '';
+  var photoUrl = '';
+  var notes = '';
+  if (data.editing === null) {
+    title = $journalEntry.elements.title.value;
+    photoUrl = $journalEntry.elements.photoUrl.value;
+    notes = $journalEntry.elements.notes.value;
+    entryObj.title = title;
+    entryObj.photoUrl = photoUrl;
+    entryObj.notes = notes;
+    entryObj.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    data.entries.unshift(entryObj);
+    $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+    $entryList.prepend(renderEntry(entryObj));
+    $journalEntry.reset();
+  } else if (data.editing !== null) {
+    title = $journalEntry.elements.title.value;
+    photoUrl = $journalEntry.elements.photoUrl.value;
+    notes = $journalEntry.elements.notes.value;
+    entryObj.title = title;
+    entryObj.photoUrl = photoUrl;
+    entryObj.notes = notes;
+    entryObj.entryId = data.editing.entryId;
+    data.entries[data.entries.length - data.editing.entryId] = entryObj;
```

Comment:
You should use push. You cannot use the id and subtraction to calculate the index. It won't work when the user deletes entries.

Reply:
I've replaced 
data.entries[data.entries.length - data.editing.entryId] = entryObj;
witth a for loop to find the data.entries object that matches the entryObj by the entryId